components/renku_data_services/message_queue/orm.py

```diff
@@ -71,3 +73,20 @@ def get_message_type(self) -> Optional[str]:
             return None
         else:
             return message_type
+
+
+class ReprovisioningORM(BaseORM):
+    """Reprovisioning table.
+
+    This table is used to make sure that only one instance of reprovisioning is run at any given time.
+    It gets updated with the reprovisioning progress.
+    """
+
+    __tablename__ = "reprovisioning"
+
+    id: Mapped[ULID] = mapped_column("id", ULIDType, primary_key=True, default_factory=lambda: str(ULID()), init=False)
+    start_date: Mapped[datetime] = mapped_column("start_date", DateTime(timezone=True), nullable=False)
```

Comment:
question: How do we handle errors during reprovisioning? And we have 2 kind here:
1. Errors that fully crash the whole reprovisioning process
2. Errors that may result from trying to reprovision a specific entity only

At least for no 1 we should save something in the database and report it back to the admin. For no 2 the question is whether we want to keep going if we encounter errors with some entities or do we fully stop?

And related to this is another question. What do we do if we encounter an error? Should we try to delete the / cleanup the events that were added so that we can effectively rollback to the pre-reprovisioining request state?